src/nirfsg/system_tests/test_system_nirfsg.py

```diff
@@ -0,0 +1,559 @@
+        assert script_exists is True
+        script_exists = test_rfsg_device_session.check_if_script_exists('myScript2')
+        assert script_exists is False
+
+    @pytest.mark.skipif(use_simulated_session is True, reason="Scripts not compiled on simulated device")
+    def test_write_script_with_bad_script(self, test_rfsg_device_session):
+        test_rfsg_device_session.configure_generation_mode(nirfsg.GenerationMode.SCRIPT)
+        script = '''script myScript1
+        repeat forever
+        generate mywaveform1
+        end repeat
+        end script'''
+        try:
+            test_rfsg_device_session.write_script(script)
+            assert False
+        except nirfsg.Error as e:
+            assert e.code == -1074101603
+            assert "A waveform matching the provided name does not exist in memory" in e.description
+
+    def test_configure_software_trigger(self, test_rfsg_device_session):
+        test_rfsg_device_session.configure_software_start_trigger()
+        assert test_rfsg_device_session.start_trigger_type == nirfsg.StartTrigType.SOFTWARE
+        test_rfsg_device_session.configure_software_script_trigger('scriptTrigger0')
+        assert test_rfsg_device_session.script_triggers[0].script_trigger_type == nirfsg.ScriptTrigType.SOFTWARE
+
+    def test_configure_digital_edge_trigger(self, test_rfsg_device_session):
+        test_rfsg_device_session.configure_digital_edge_start_trigger('PXI_Trig1', nirfsg.StartTrigDigEdgeEdge.RISING)
+        test_rfsg_device_session.configure_digital_edge_script_trigger('scriptTrigger1', 'PXI_Trig2', nirfsg.ScriptTrigDigEdgeEdge.FALLING)
+        assert test_rfsg_device_session.start_trigger_type == nirfsg.StartTrigType.DIGITAL_EDGE
+        assert test_rfsg_device_session.digital_edge_start_trigger_source == 'PXI_Trig1'
+        assert test_rfsg_device_session.digital_edge_start_trigger_edge == nirfsg.StartTrigDigEdgeEdge.RISING
+        assert test_rfsg_device_session.script_triggers[1].script_trigger_type == nirfsg.ScriptTrigType.DIGITAL_EDGE
+        assert test_rfsg_device_session.script_triggers[1].digital_edge_script_trigger_source == 'PXI_Trig2'
+        assert test_rfsg_device_session.script_triggers[1].digital_edge_script_trigger_edge == nirfsg.ScriptTrigDigEdgeEdge.FALLING
+
+    def test_disable_trigger(self, test_rfsg_device_session):
+        test_rfsg_device_session.configure_software_start_trigger()
+        assert test_rfsg_device_session.start_trigger_type == nirfsg.StartTrigType.SOFTWARE
+        test_rfsg_device_session.disable_start_trigger()
+        assert test_rfsg_device_session.start_trigger_type == nirfsg.StartTrigType.NONE
+        test_rfsg_device_session.configure_software_script_trigger('scriptTrigger3')
+        assert test_rfsg_device_session.script_triggers[3].script_trigger_type == nirfsg.ScriptTrigType.SOFTWARE
+        test_rfsg_device_session.disable_script_trigger('scriptTrigger3')
```

Comment:
> disable_script_trigger('scriptTrigger3')

The trigger_id shouldn't be a parameter, but a repeated capability. We have a script_triggers rep cap.